### PR TITLE
Add Logout Content Negotiation

### PR DESCRIPTION
**Pull request by rwinch**
Adds Logout Content Negotiation

Fixes gh-3282

### CODE REVIEW

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java [hunk 1]
```diff
@@ -206,6 +212,25 @@ public LogoutConfigurer() {
...
+	 * {@link SimpleUrlLogoutSuccessHandler}.
+	 *
+	 * @param handler the {@link LogoutSuccessHandler} to use
+	 * @param preferredMatcher the {@link RequestMatcher} for this default
+	 * {@link LogoutSuccessHandler}
+	 * @return the {@link LogoutConfigurer} for further customizations
+	 */
+	public LogoutConfigurer<H> defaultLogoutSuccessHandlerFor(
+			LogoutSuccessHandler handler, RequestMatcher preferredMatcher) {
+		this.defaultLogoutSuccessHandlerMappings.put(preferredMatcher, handler);
```
Comment (jgrandja) [inline config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java]: Should we add null assertion checks to both arguments?

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java [hunk 2]
```diff
@@ -206,6 +212,25 @@ public LogoutConfigurer() {
 	}
 
 	/**
+	 * Sets a default {@link LogoutSuccessHandler} to be used which prefers being invoked
+	 * for the provided {@link RequestMatcher}. If no {@link LogoutSuccessHandler} is
+	 * specified a {@link SimpleUrlLogoutSuccessHandler} will be used.
+	 * If any default {@link LogoutSuccessHandler} instances are configured, then a
+	 * {@link DelegatingAuthenticationEntryPoint} will be used that defaults to a
```
Comment (jgrandja) [inline config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java]: Typo: `DelegatingAuthenticationEntryPoint` should be `DelegatingLogoutSuccessHandler`

File: web/src/test/java/org/springframework/security/web/authentication/logout/DelegatingLogoutSuccessHandlerTests.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+/**
+ * RequestMatcherLogoutSuccessHandler Tests
```
Comment (jgrandja) [inline web/src/test/java/org/springframework/security/web/authentication/logout/DelegatingLogoutSuccessHandlerTests.java]: Should be `DelegatingLogoutSuccessHandlerTests`

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java (hunk 1 above)
Comment (rwinch) [inline config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java]: Fixed

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java (hunk 2 above)
Comment (rwinch) [inline config/src/main/java/org/springframework/security/config/annotation/web/configurers/LogoutConfigurer.java]: Fixed

File: web/src/test/java/org/springframework/security/web/authentication/logout/DelegatingLogoutSuccessHandlerTests.java (hunk 3 above)
Comment (rwinch) [inline web/src/test/java/org/springframework/security/web/authentication/logout/DelegatingLogoutSuccessHandlerTests.java]: Fixed

**Pull request merged**